tests/integration_test.py

```diff
@@ -101,6 +103,13 @@ def tearDown(self):
                 pass
         for folder in self.tmp_folders:
             shutil.rmtree(folder)
+
+        for volume in self.tmp_volumes:
+            try:
+                self.client.remove_volume(volume)
+            except docker.errors.APIError:
+                pass
```

Comment:
Isn't an iteration on a list of size 0 almost a no-op? It feels to me like we would make the code more complex without actually gaining anything from it.